robot_smach_states/test/designators/empty_spot_designator.py

```diff
@@ -1,12 +1,15 @@
 #! /usr/bin/env python
 
+import sys
 import rospy
 import robot_smach_states.util.designators as ds
+from robot_skills import get_robot
 
 if __name__ == "__main__":
     rospy.init_node("testdesignator")
-    from robot_skills.amigo import Amigo
-    robot = Amigo()
+
+    robot_name = sys.argv[1] if len(sys.argv) > 1 else "amigo"
+    robot = get_robot(robot_name)
```

Comment:
I wonder if the default robot selection should also be inside `get_robot`. Current solution is more code, but has the nice property that the default is selectable at every call.

Alternatively, move argument selection into a new function?
```` python
import sys

def get_robot_from_argv(index, default_robot_name='amigo'):
    """
    Construct a robot from the name given in the command-line or from the default robot name.
    :param index: Index in the command-line arguments where a robot name may be available.
    :param default_robot_name: Name of the robot to use if the command line did not contain a name.
    :return: The constructed robot.
    :raise: RunTimeError if no robot could be created.
    """
    if len(sys.argv) > index:
        robot_name = sys.argv[index]
    else:
        robot_name = default_robot_name

    return get_robot(robot_name)
````

Reply:
See a71172c306547eba6d50c057baaef8c0ec10a1d0